File: javascript/src/flipInt.js
/*
  Write a function that takes a positive integer and reverses it. Do so
  without string manipulation, arrays, or bitwise operators.

  Example:
    An input of 123 would return 321
*/
const flipInt = (n) => {
  let temp = n,
        num = 0

  for (let i = Math.floor(Math.log10(n)); i >= 0; i--) {
    num += 10**i * (temp % 10)
    temp = Math.floor(temp/10)
  }
  return num
}

module.exports = flipInt

File: python/src/flip_int.py
#!/usr/bin/env python
import math

def flip_int(x):
    temp = math.fabs(x)
    length = math.ceil(math.log10(temp+1) or 1)
    sign = math.copysign(1, x)

    place = 10**(length-1)
    num = 0
    for i in range(0,length):
        num += temp%10 * place
        temp = math.floor(temp/10)
        place = math.floor(place/10)
    return int(num * sign)


def main():
    print(flip_int(123))
    print(flip_int(-321))


if __name__ == '__main__':
    main()
